### ISSUE TITLE
D417: Ignore arguments marked as unused

### ISSUE THREAD

**Issue by karlch** [PyCQA/pydocstyle#438]
First of all thanks for a really useful tool and the great D417 which has caught many issues for me!

I feel like D417 should not complain for arguments marked as unused by prepending an underscore. This is also how pylint handles the `unused-argument` error.

Consider e.g.:
```
def handle_interrupt(self, signum: int, _frame: FrameType) -> None:
    ...
```

We provide the `_frame` argument to make the callback signature valid, but never actually use it in the later code. IMHO documenting `_frame` in this case should not be required.
